File: src/main/php/Easy/StringIsSubsequence.php
<?php

declare(strict_types=1);

namespace Src\Easy;

class StringIsSubsequence
{
	   /**
     * @param String $s
     * @param String $t
     * @return Boolean
     */
    public function transformStringIsSubsequence(string $s, string $t): bool
    {
        $left = 0;
        $right = 0;
        while ($left < strlen($s) && $right < strlen($t)) {
            if ($s[$left] === $t[$right]) {
                $left++;
            }
            $right++;
        }

        return $left === strlen($s);
    }

    public function solve(string $s, string $t): bool
    {
        return $this->transformStringIsSubsequence($s, $t);
    }
}

File: easy/string_is_subsequence.go
package easy

func StringIsSubsequence(s string, t string) bool {
	left := 0
	right := 0
	for left < len(s) && right < len(t) {
		if s[left] == t[right] {
			left++
		}
		right++
	}

	return left == len(s)
}
